R/drop_duplicate_tips_random.R

```diff
@@ -0,0 +1,78 @@
+#' If a language tree has tips with duplicate tip labels, drop all but one at random. If there are tips which are dialects of the same language, you can choose to drop all but one.
+#'
+#' @param tree 	an object of class "phylo" with glottocodes in tip labels.
+#' @param merge_dialects a logical specifying whether to replace dialect tip labels with the glottocode of the language that is their parent, and then drop all but one
+#' @param random_seed integer specifying random seed
+#' @param LanguageTable data-frame of a cldf LanguageTable with columns for glottocodes per languoid and dialects language parent glottocode. If merge_dialects is set to TRUE, a LanguageTable is necessary.
+#' @param trim_tip_label_to_first_eight a logical specifying whether we should trim the tip label to the first 8 characters. If the tip labels contain more than just glottocodes, it is necessary to trim away what is not glottocodes. If set to TRUE, only the first 8 characters are retained. This is necessary for the EDGE-trees as they contain names as well (e.g. "bamu1257_Bamu_Kiwaian"). If the tip labels only consist of glottocodes, this parameter can be set to TRUE or FALSE without difference.
```

Comment:
I think the functionality to rename tips doesn't belong here. It's a one-liner to do that before calling this function. Typically, tree-manipulation packages will also have more flexible functionality to do this, see e.g.
https://github.com/dlce-eva/python-newick/blob/b16cc4b93a8fd246b7500f366f2d35eb8ff10f70/src/newick.py#L352-L359
or
https://github.com/dlce-eva/python-newick/blob/b16cc4b93a8fd246b7500f366f2d35eb8ff10f70/src/newick.py#L334-L345

Reply:
For rgrambank, I'm specifically thinking about analysis we do in grambank code a lot and it's often with the EDGE-tree. The EDGE-tree tips need to be renamed, and I think it'd be convenient to do this in one sweep.

@SimonGreenhill your call. Another hill I don't need to die on.